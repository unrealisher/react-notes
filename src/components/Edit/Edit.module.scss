.list {
  display: flex;
}

.item:not(:last-child) {
  margin-right: 16px;
}

.button {
  cursor: pointer;
  background-color: inherit;

  &:hover .path,
  &:focus .path {
    fill: rgba(0, 0, 0, 1);
  }

  &:active .path {
    fill: rgba(0, 0, 0, 0.8);
  }
}

@media (min-width: 704px) {
  .list {
    visibility: hidden;
  }
}
