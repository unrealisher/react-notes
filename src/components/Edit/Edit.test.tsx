import React from "react";
import { shallow, mount } from "enzyme";
import { Edit } from "./Edit";

describe("Edit", () => {
  const onArchiveClick = (): void => {};
  const note = {
    id: 6,
    type: "text",
    text:
      "Реферат до 2 апреля (300 терминов, аннотация, основная, заключение, список литературы)\nПеревод 40к знаков к зачету, 40к к экзамену\nТест 1 и 2\n\nПодготовка кадров зарубежом (юнит 2)\nНаучно исследовательская работа магистранта (юнит 2)\nРоль науки (юнит 1)\n10-15 предложений минимум",
    tags: [5, 6],
    color: 1,
    size: "m",
    created: 1549189500000
  };
  const setPatchItem = (): void => {};
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <Edit
        onArchiveClick={onArchiveClick}
        note={note}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("onArchiveClick calls when first button is clicked", () => {
    const onArchiveClick = jest.fn();
    const component = mount(
      <Edit
        onArchiveClick={onArchiveClick}
        note={note}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    const button = component.find(".button").first();
    button.simulate("click");
    expect(onArchiveClick).toBeCalledWith(note.id);
  });

  it("setPatchItem calls when second button is clicked", () => {
    const setPatchItem = jest.fn();
    const component = mount(
      <Edit
        onArchiveClick={onArchiveClick}
        note={note}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    const button = component.find(".button").last();
    button.simulate("click");
    expect(setPatchItem).toBeCalledWith(note);
  });

  it("setPopup calls when second button is clicked", () => {
    const setPopup = jest.fn();
    const component = mount(
      <Edit
        onArchiveClick={onArchiveClick}
        note={note}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    const button = component.find(".button").last();
    button.simulate("click");
    expect(setPopup).toBeCalledWith(true);
  });
});
