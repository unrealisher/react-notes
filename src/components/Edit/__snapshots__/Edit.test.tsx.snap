// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Edit
    note={
      Object {
        "color": 1,
        "created": 1549189500000,
        "id": 6,
        "size": "m",
        "tags": Array [
          5,
          6,
        ],
        "text": "Реферат до 2 апреля (300 терминов, аннотация, основная, заключение, список литературы)
Перевод 40к знаков к зачету, 40к к экзамену
Тест 1 и 2

Подготовка кадров зарубежом (юнит 2)
Научно исследовательская работа магистранта (юнит 2)
Роль науки (юнит 1)
10-15 предложений минимум",
        "type": "text",
      }
    }
    onArchiveClick={[Function]}
    setPatchItem={[Function]}
    setPopup={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="item"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            <svg
              className="svg"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                fill="rgba(0, 0, 0, 0.3)"
              />
            </svg>
          </button>
        </li>,
        <li
          className="item"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            <svg
              className="svg"
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                clipRule="evenodd"
                d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                fill="rgba(0, 0, 0, 0.3)"
                fillRule="evenodd"
              />
            </svg>
          </button>
        </li>,
      ],
      "className": "list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="button"
            onClick={[Function]}
          >
            <svg
              className="svg"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                fill="rgba(0, 0, 0, 0.3)"
              />
            </svg>
          </button>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="svg"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                fill="rgba(0, 0, 0, 0.3)"
              />
            </svg>,
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <path
                className="path"
                d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                fill="rgba(0, 0, 0, 0.3)"
              />,
              "className": "svg",
              "fill": "none",
              "height": "16",
              "viewBox": "0 0 16 16",
              "width": "16",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "path",
                "d": "M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z",
                "fill": "rgba(0, 0, 0, 0.3)",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            "type": "svg",
          },
          "type": "button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="button"
            onClick={[Function]}
          >
            <svg
              className="svg"
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                clipRule="evenodd"
                d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                fill="rgba(0, 0, 0, 0.3)"
                fillRule="evenodd"
              />
            </svg>
          </button>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="svg"
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="path"
                clipRule="evenodd"
                d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                fill="rgba(0, 0, 0, 0.3)"
                fillRule="evenodd"
              />
            </svg>,
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <path
                className="path"
                clipRule="evenodd"
                d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                fill="rgba(0, 0, 0, 0.3)"
                fillRule="evenodd"
              />,
              "className": "svg",
              "fill": "none",
              "height": "14",
              "viewBox": "0 0 14 14",
              "width": "14",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "path",
                "clipRule": "evenodd",
                "d": "M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z",
                "fill": "rgba(0, 0, 0, 0.3)",
                "fillRule": "evenodd",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            "type": "svg",
          },
          "type": "button",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              <svg
                className="svg"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                  fill="rgba(0, 0, 0, 0.3)"
                />
              </svg>
            </button>
          </li>,
          <li
            className="item"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              <svg
                className="svg"
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  clipRule="evenodd"
                  d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                  fill="rgba(0, 0, 0, 0.3)"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </li>,
        ],
        "className": "list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button"
              onClick={[Function]}
            >
              <svg
                className="svg"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                  fill="rgba(0, 0, 0, 0.3)"
                />
              </svg>
            </button>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="svg"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                  fill="rgba(0, 0, 0, 0.3)"
                />
              </svg>,
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <path
                  className="path"
                  d="M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z"
                  fill="rgba(0, 0, 0, 0.3)"
                />,
                "className": "svg",
                "fill": "none",
                "height": "16",
                "viewBox": "0 0 16 16",
                "width": "16",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "path",
                  "d": "M14.623 0L5.473 13.174L1.186 8.574L0 10.03L5.62 16L16 1.27L14.623 0Z",
                  "fill": "rgba(0, 0, 0, 0.3)",
                },
                "ref": null,
                "rendered": null,
                "type": "path",
              },
              "type": "svg",
            },
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button"
              onClick={[Function]}
            >
              <svg
                className="svg"
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  clipRule="evenodd"
                  d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                  fill="rgba(0, 0, 0, 0.3)"
                  fillRule="evenodd"
                />
              </svg>
            </button>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="svg"
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="path"
                  clipRule="evenodd"
                  d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                  fill="rgba(0, 0, 0, 0.3)"
                  fillRule="evenodd"
                />
              </svg>,
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <path
                  className="path"
                  clipRule="evenodd"
                  d="M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z"
                  fill="rgba(0, 0, 0, 0.3)"
                  fillRule="evenodd"
                />,
                "className": "svg",
                "fill": "none",
                "height": "14",
                "viewBox": "0 0 14 14",
                "width": "14",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "path",
                  "clipRule": "evenodd",
                  "d": "M2.561 13.9985H0.319242C0.142929 13.9985 0 13.8596 0 13.6793V11.4375C0 11.2612 0.100634 11.0176 0.221503 10.8968L8.73926 2.379C8.86159 2.25667 9.06327 2.26001 9.18677 2.3835L11.615 4.81173C11.7398 4.93655 11.7404 5.13837 11.6195 5.25924L3.10175 13.777C2.97942 13.8993 2.74133 13.9985 2.561 13.9985ZM13.0238 3.85498C12.9038 3.97491 12.6999 3.97647 12.5751 3.85165L10.1469 1.42342C10.0234 1.29992 10.0205 1.09772 10.1435 0.974739L10.8809 0.237332C11.2138 -0.0756243 11.5334 -0.082578 11.841 0.237332L13.7612 2.1575C14.0849 2.47776 14.074 2.80892 13.7612 3.11758L13.0238 3.85498Z",
                  "fill": "rgba(0, 0, 0, 0.3)",
                  "fillRule": "evenodd",
                },
                "ref": null,
                "rendered": null,
                "type": "path",
              },
              "type": "svg",
            },
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
